Contents:
01 Import libraries and data
02 Check Output
03 Aggregate Key Values
04 Run Consistency Checks


01 Import libraries and data

In [2]:
import pandas as pd
import numpy as np
import os

In [3]:
df = pd.read_csv(r'/Users/Owner/Documents/CareerFoundry/Data Immersion/Achievement 6/Mental Health Data.csv', index_col = False)

02 Check Output

In [5]:
df.head()

Entity Code  Year      Country, Year  Schizophrenia (%)  \
0  Afghanistan  AFG  1990  Afghanistan, 1990           0.160560   
1  Afghanistan  AFG  1991  Afghanistan, 1991           0.160312   
2  Afghanistan  AFG  1992  Afghanistan, 1992           0.160135   
3  Afghanistan  AFG  1993  Afghanistan, 1993           0.160037   
4  Afghanistan  AFG  1994  Afghanistan, 1994           0.160022   

   Bipolar disorder (%)  Eating disorders (%)  Anxiety disorders (%)  \
0              0.697779              0.101855               4.828830   
1              0.697961              0.099313               4.829740   
2              0.698107              0.096692               4.831108   
3              0.698257              0.094336               4.830864   
4              0.698469              0.092439               4.829423   

   Drug use disorders (%)  Depression (%)  ...  Age 50-69   Age 70+  \
0                1.677082        4.071831  ...   5.917752  5.202803   
1                1.684746        4.079531  ...   5.927093  5.192849   
2                1.694334        4.088358  ...   5.945656  5.176872   
3                1.705320        4.096190  ...   5.966915  5.167355   
4                1.716069        4.099582  ...   5.975907  5.157549   

   Age 15-49  Age Standardized  Prevalence in Males  Prevalence in Females  \
0   4.939766          4.071831             3.499982               4.647815   
1   4.902682          4.079531             3.503947               4.655772   
2   4.837097          4.088358             3.508912               4.662066   
3   4.813657          4.096190             3.513429               4.669012   
4   4.839340          4.099582             3.515578               4.673050   

   Population  Suicide rate (deaths per 100,000 individuals)  \
0    12412000                                      10.318504   
1    13299000                                      10.327010   
2    14486000                                      10.271411   
3    15817000                                      10.376123   
4    17076000                                      10.575915   

   Depressive disorder rates (number suffering per 100,000)  \
0                                        4039.755763          
1                                        4046.256034          
2                                        4053.709902          
3                                        4060.203474          
4                                        4062.290365          

   Number with depression  
0             318435.8137  
1             329044.7740  
2             382544.5729  
3             440381.5074  
4             456916.6455  

[5 rows x 27 columns]

In [6]:
df.columns

Index(['Entity', 'Code', 'Year', 'Country, Year', 'Schizophrenia (%)',
       'Bipolar disorder (%)', 'Eating disorders (%)', 'Anxiety disorders (%)',
       'Drug use disorders (%)', 'Depression (%)', 'Alcohol use disorders (%)',
       'All Ages', 'Age 10-14', 'Age 15-19', 'Age 20-24', 'Age 25-29',
       'Age 30-34', 'Age 50-69', 'Age 70+', 'Age 15-49', 'Age Standardized',
       'Prevalence in Males', 'Prevalence in Females', 'Population',
       'Suicide rate (deaths per 100,000 individuals)',
       'Depressive disorder rates (number suffering per 100,000)',
       'Number with depression'],
      dtype='object')

03 Aggregate Key Values

In [8]:
pd.set_option('display.max_columns', None)

In [9]:
df.describe()

Year  Schizophrenia (%)  Bipolar disorder (%)  \
count  6468.000000        6468.000000           6468.000000   
mean   2003.500000           0.211644              0.719145   
std       8.078372           0.044253              0.171589   
min    1990.000000           0.146902              0.314535   
25%    1996.750000           0.181530              0.615532   
50%    2003.500000           0.199563              0.693134   
75%    2010.250000           0.236365              0.835063   
max    2017.000000           0.375110              1.206597   

       Eating disorders (%)  Anxiety disorders (%)  Drug use disorders (%)  \
count           6468.000000            6468.000000             6468.000000   
mean               0.239998               3.989921                0.862278   
std                0.158141               1.167526                0.460679   
min                0.073908               2.023393                0.383650   
25%                0.122387               3.188824                0.535064   
50%                0.182525               3.554373                0.726430   
75%                0.292666               4.682163                0.940157   
max                0.943991               8.967330                3.452476   

       Depression (%)  Alcohol use disorders (%)     All Ages    Age 10-14  \
count     6468.000000                6468.000000  6468.000000  6468.000000   
mean         3.497654                   1.585821     3.279958     1.376053   
std          0.655859                   0.860283     0.861814     0.404854   
min          2.139903                   0.446940     1.805876     0.710318   
25%          3.005529                   0.993685     2.609811     1.066155   
50%          3.499606                   1.479937     3.031299     1.259560   
75%          3.912381                   1.867834     3.793446     1.589738   
max          6.602754                   5.474668     6.990122     3.303273   

         Age 15-19    Age 20-24    Age 25-29    Age 30-34    Age 50-69  \
count  6468.000000  6468.000000  6468.000000  6468.000000  6468.000000   
mean      3.076291     3.788878     3.913513     4.076232     5.662794   
std       0.857978     1.075230     1.026875     1.030536     1.020436   
min       1.497964     1.718949     1.946369     2.231433     3.248216   
25%       2.529356     3.071424     3.172756     3.248436     4.862645   
50%       2.895936     3.528323     3.722521     3.969812     5.621947   
75%       3.415008     4.309666     4.488815     4.725525     6.416751   
max       9.053459    11.275973    10.377815     9.684816     9.777747   

           Age 70+    Age 15-49  Age Standardized  Prevalence in Males  \
count  6468.000000  6468.000000       6468.000000          6468.000000   
mean      6.137538     4.016824          3.497654             2.806911   
std       1.501730     0.967864          0.655859             0.534332   
min       3.249315     2.177050          2.139903             1.594129   
25%       4.874158     3.316308          3.005529             2.412512   
50%       5.928127     3.850302          3.499606             2.779003   
75%       7.361696     4.597567          3.912381             3.138412   
max      11.531880     9.626314          6.602754             4.905163   

       Prevalence in Females    Population  \
count            6468.000000  6.468000e+03   
mean                4.156787  5.717406e+07   
std                 0.840319  4.390243e+08   
min                 2.348094  0.000000e+00   
25%                 3.539870  3.437500e+05   
50%                 4.199842  4.753000e+06   
75%                 4.649112  1.800525e+07   
max                 8.537801  7.547859e+09   

       Suicide rate (deaths per 100,000 individuals)  \
count                                    6468.000000   
mean                                       12.068240   
std                                         8.037739   
min                                         1.526792   
25%        

04 Run Consistency Checks

In [19]:
#check for mixed-type data

for col in df.columns.tolist():
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
  if len (df[weird]) > 0:
    print (col)

/var/folders/ys/d72m3mmn3kxcfr0s6frmm81w0000gn/T/ipykernel_83924/2587165438.py:4: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
/var/folders/ys/d72m3mmn3kxcfr0s6frmm81w0000gn/T/ipykernel_83924/2587165438.py:4: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
/var/folders/ys/d72m3mmn3kxcfr0s6frmm81w0000gn/T/ipykernel_83924/2587165438.py:4: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
/var/folders/ys/d72m3mmn3kxcfr0s6frmm81w0000gn/T/ipykernel_83924/2587165438.py:4: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
/var/fol

Code


/var/folders/ys/d72m3mmn3kxcfr0s6frmm81w0000gn/T/ipykernel_83924/2587165438.py:4: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
/var/folders/ys/d72m3mmn3kxcfr0s6frmm81w0000gn/T/ipykernel_83924/2587165438.py:4: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
/var/folders/ys/d72m3mmn3kxcfr0s6frmm81w0000gn/T/ipykernel_83924/2587165438.py:4: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
/var/folders/ys/d72m3mmn3kxcfr0s6frmm81w0000gn/T/ipykernel_83924/2587165438.py:4: FutureWarning: DataFrame.applymap has been deprecated. Use DataFrame.map instead.
  weird = (df[[col]].applymap(type) != df[[col]].iloc[0].apply(type)).any(axis = 1)
/var/fol

The check returned no resuts, suggesting there is no mixed-type data.

In [22]:
#check for missing values
df.isnull().sum()

Entity                                                        0
Code                                                        980
Year                                                          0
Country, Year                                                 0
Schizophrenia (%)                                             0
Bipolar disorder (%)                                          0
Eating disorders (%)                                          0
Anxiety disorders (%)                                         0
Drug use disorders (%)                                        0
Depression (%)                                                0
Alcohol use disorders (%)                                     0
All Ages                                                      0
Age 10-14                                                     0
Age 15-19                                                     0
Age 20-24                                                     0
Age 25-29                               

980 "entities" have no country code. I do not expect to use this field.

In [25]:
#check for duplicates
df_dups = df[df.duplicated()]

In [27]:
df_dups

Empty DataFrame
Columns: [Entity, Code, Year, Country, Year, Schizophrenia (%), Bipolar disorder (%), Eating disorders (%), Anxiety disorders (%), Drug use disorders (%), Depression (%), Alcohol use disorders (%), All Ages, Age 10-14, Age 15-19, Age 20-24, Age 25-29, Age 30-34, Age 50-69, Age 70+, Age 15-49, Age Standardized, Prevalence in Males, Prevalence in Females, Population, Suicide rate (deaths per 100,000 individuals), Depressive disorder rates (number suffering per 100,000), Number with depression]
Index: []

As I made no changes in Jupyter, there is no need to export data.